docs/bundler/migration.md

```diff
@@ -0,0 +1,657 @@
+Bun's bundler API is inspired heavily by [esbuild](https://esbuild.github.io/). Migrating to Bun's bundler from esbuild should be relatively painless. This guide will briefly explain why you might consider migrating to Bun's bundler and provide a side-by-side API comparison reference for those who are already familiar with esbuild's API.
+
+## Performance
+
+This is the simplest reason to migrate to Bun's bundler. Bun borrowed heavily from esbuild's design while avoiding some architectural issues that limit its maximum performance. Plus, Bun's bundler takes full advantage of the extensive optimization work that's been done on Bun's internal JS/TS parser.
+
+In the end,
+
+## CLI
+
+Bun and esbuild both provide a command-line interface.
+
+```bash
+$ esbuild <entrypoint> --bundle --outdir=out --bundle
+$ bun build <entrypoint> --outdir=out
+```
+
+There are a few behavioral differences to note.
+
+- Bun _always bundles by default_. This is why the `--bundle` flag isn't necessary in the Bun example.
+- Bun's bundler does not include a built-in development server or file watcher. It's just a bundler. The bundler can be used in conjunction with `Bun.serve`, `--watch`, and other built-in Bun functionality to achieve the same effect. As such, all options relating to HTTP/file watching are not applicable.
+
+{% table %}
+
+- `esbuild`
+- `bun build`
+
+---
+
+- `--bundle`
+- n/a
+- Not necessary, `bun build` always bundles.
+
+---
+
+- `--define:K=V`
+- `--define:K=V`
+- No differences
+
+---
+
+- `--external`
+- `--external`
```

Comment:
instead of `--external:module`, we use `--external module` or `--external=module`.

we don't do that `:` syntax anywhere, though the reason they chose that is pretty interesting (it's for any options that can be repeated, like array) and maybe something we should consider